core/store.go

```diff
@@ -197,3 +197,28 @@ func countClients(clients *sync.Map) int {
 	})
 	return count
 }
+
+func GetNoTouch(k string) *Obj {
+	storeMutex.RLock()
+	defer storeMutex.RUnlock()
+	keypoolMutex.RLock()
+	defer keypoolMutex.RUnlock()
+
+	ptr, ok := keypool[k]
+	if !ok {
+		return nil
+	}
+
+	v := store[ptr]
+	if v != nil {
+		if hasExpired(v) {
+			keypoolMutex.RUnlock()
+			storeMutex.RUnlock()
+			Del(k)
```

Comment:
We shouldn't be deleting a key in a `NoTouch` function. There are other functions which will check the clear expired keys.